src/snippetsStorage.ts

```diff
@@ -184,31 +189,45 @@ export default class SnippetsStorage {
     this.elements.set(data.id, element);
     this.getElement(parentId).childIds?.push(data.id);
 
-    await this.save();
+    await this.save("save snippet");
     this.onSnippetSave?.(element);
   }
 
   getSnippet(id: string): string {
     return this.getElement(id).data.content?.toString() || "";
   }
 
-  async save(): Promise<void> {
+  async save(operation?: string): Promise<void> {
```

Comment:
I wonder if `operation` could be an enum. This way, we can the operations get type checked and spelling mistakes get prevented.